src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -233,6 +233,11 @@ public partial class EventSource : IDisposable
 #pragma warning restore CA1823
 #endif //FEATURE_EVENTSOURCE_XPLAT
 
+        private static bool IsSupported { get; } = InitializeIsSupported();
+
+        private static bool InitializeIsSupported() =>
+            AppContext.TryGetSwitch("System.Diagnostics.Tracing.EventSource.IsSupported", out bool isSupported) ? isSupported : true;
```

Comment:
We should add some tests that set this switch and validate EventSource usage doesn't blow up in some unexpected way.

Reply:
I've hit an issue trying to set this switch in a test.

The problem is that EventSource gets initialized before any managed code can run, so I can't set the `AppContext` switch in my test.

I've tried setting the backing field through reflection, but the runtime doesn't like that:

```
System.FieldAccessException: Cannot set initonly static field '<IsSupported>k__BackingField' after type 'System.Diagnostics.Tracing.EventSource' is initialized.
```

And from what I can tell, RemoteExecutor doesn't allow for injecting runtimeconfig settings into the process.

Options I can think of to fix this are:

1. Add support for an environment variable (similar to the way we define GlobalizationMode.Invariant).
2. Add support into RemoteExecutor for adding runtimeconfig settings.
3. Change the code pattern here to not use a readonly field - but that might have perf implications for the normal code execution when this switch isn't set.

Any others you can think of?

I think (2) would be the most correct - but it is expensive to write. (1) would be pretty cheap, and it would be symmetrical with how we define GlobalizationMode.Invariant, but it would mean another mechanism we would need to support.

Reply:
You can build a regular .exe without the RemoteExecutor baggage. It won't be the first one in the tree. There are other situations where the RemoteExecutor is getting into way:

E.g. https://github.com/dotnet/runtime/tree/master/src/libraries/System.Threading.Thread/tests/STAMain

Reply:
If we can augment RemoteExecutor to support this, that'd be my preference.  If it proves too cumbersome or problematic, we could fall back to a dedicated exe.

Reply:
I have a PR for the RemoteExecutor change here: https://github.com/dotnet/arcade/pull/5698